File: SWB4/Client/OpenOffice/OfficeComunicationInterfaces/src/org/semanticwb/office/interfaces/VersionInfo.java
/*
 * To change this template, choose Tools | Templates
 * and open the template in the editor.
 */

package org.semanticwb.office.interfaces;

import java.util.Date;

/**
 *
 * @author victor.lorenzana
 */
public class VersionInfo {

    @Override
    public boolean equals(Object obj)
    {
        if (obj == null)
        {
            return false;
        }
        if (getClass() != obj.getClass())
        {
            return false;
        }
        final VersionInfo other = (VersionInfo) obj;
        if ((this.nameOfVersion == null) ? (other.nameOfVersion != null) : !this.nameOfVersion.equals(other.nameOfVersion))
        {
            return false;
        }
        return true;
    }

    @Override
    public int hashCode()
    {
        int hash = 3;
        hash = 73 * hash + (this.nameOfVersion != null ? this.nameOfVersion.hashCode() : 0);
        return hash;
    }

    

    @Override
    public String toString()
    {
        if(nameOfVersion.equals("*"))
        {
            return "Mostrar la última version";
        }
        return nameOfVersion;
    }

    public boolean published;
    public String contentId;    
    public String nameOfVersion;
    public Date created;
    public String user;
}

File: SWB4/Client/Microsoft Office/INFOTEC WebBuilder 4/WBOffice4/Interfaces/VersionInfo.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Xml;
using System.Xml.Serialization;
namespace WBOffice4.Interfaces
{
    
    public class VersionInfo
    {
    
        public String contentId;
    
        public String nameOfVersion;
    
        public DateTime created;
    
        public String user;

        public bool published;

        public override String ToString()
        {
            if (nameOfVersion == "*")
            {
                return "Mostrar la última version";
            }
            else
            {
                return nameOfVersion;

            }
        }
        public override bool Equals(Object obj)
        {
            if (obj == null)
            {
                return false;
            }
            if (GetType() != obj.GetType())
            {
                return false;
            }
            VersionInfo other = (VersionInfo)obj;
            if ((this.nameOfVersion == null) ? (other.nameOfVersion != null) : !this.nameOfVersion.Equals(other.nameOfVersion))
            {
                return false;
            }
            return true;
        }


        public override int GetHashCode()
        {
            int hash = 7;
            hash = 89 * hash + (this.nameOfVersion != null ? this.nameOfVersion.GetHashCode() : 0);
            return hash;
        }
    }
}
